submissions/zalevskyi/friends-app/app.js

```diff
@@ -0,0 +1,187 @@
+const USERS_DATA = []
+const USERS_QUANTITY = 30
+const RESULTS = document.getElementById('results')
+const SORT = document.getElementById('sort')
+const FILTER = document.getElementById('filter')
+const REQUEST_PARAMETERS = {}
+let connectionAttemps = 1
+
+function connectToServer() {
+  RESULTS.innerHTML = `<div class='connection_message'>Please wait. Connecting to server. Attemp: ${connectionAttemps}</div>`
+  connectionAttemps++
+  fetchUserData(connectToServer)
+}
+function fetchUserData(onError) {
+  let apiUrl = 'https://randomuser.me/api/1.3/'
+  let fields = ['name', 'gender', 'location', 'dob', 'phone', 'picture']
+  fetch(`${apiUrl}?results=${USERS_QUANTITY}&inc=${fields.join(',')}`, {
+    headers: { 'Content-Type': 'application/json' },
+  })
```

Comment:
Just return `fetch` and then you will have a possibility to write `fetchUserData().catch()`